clients/drcachesim/tools/filter/record_filter.cpp

```diff
@@ -213,6 +213,7 @@ record_filter_t::initialize_shard_output(per_shard_t *per_shard,
                    shard_stream->get_shard_index(), output_ext_.c_str(), version_,
                    filetype_);
             per_shard->output_path += output_ext_;
+            per_shard->filetype = static_cast<addr_t>(filetype_);
```

Comment:
Was this supposed to be part of #6722? Maybe make it part of a test?

Reply:
It should have been part of #6704 which added these global values for the synthesized headers but didn't set them in the shards.  The test modified in this PR to use opcode_mix fails without this fix.  Do you mean, try to get a unit test to hit this instead of an end-to-end test?

Reply:
Or do you mean, add an assert that the filetype is set at exit time or sthg?

Reply:
A record_filter_unit_test that verifies that filetype is set properly.

Reply:
> A record_filter_unit_test that verifies that filetype is set properly.

That is not simple: the unit test uses mocks which do not reproduce any of the desired behavior here.  The behavior we want to test requires the filetype being available at init time in some streams (those with non-empty names) but not in others.  It also requires multiple worker threads to avoid hanging on the wait.  None of that infrastructure is in the unit test file now.  Probably the easiest thing would be to create a full analyzer and scheduler with worker threads: but that is much of the way to the end-to-end test.  WDYT?

Reply:
Sure since we have the end-to-end test, this is fine.